web-portal/internal-api/upload-images.js

```diff
@@ -24,9 +19,31 @@ const IMAGE_DESTINATIONS = {
 
 const TYPES = Object.keys(IMAGE_DESTINATIONS)
 
-const s3UrlBase = process.env.AWS_SERVER_URL + process.env.AWS_S3_UPLOADS_BUCKET
+// Send images to AWS S3 only if all environment vars necessary to configure
+// are present, falling back on local upload if not
+// TODO: this check might be better refactored into a factory
+// (especially if/when we introduce support for Azure Blob Storage)
+const uploader = process.env.AWS_REGION &&
```

Comment:
you could probably make this a little easier to read by creating a function called something like getUploaderBasedOnEnv

```
function getUploaderBasedOnEnv() {
   if (process.env.AWS_REGION &&  process.env.AWS_ACCESS_KEY_ID && ...) {
      return new S3Uploader(
         process.env.AWS_SERVER_URL,
         process.env.AWS_REGION,
         process.env.AWS_ACCESS_KEY_ID,
         process.env.AWS_SECRET_ACCESS_KEY,
         process.env.AWS_S3_UPLOADS_BUCKET)
   } else {
       return new LocalUploader(process.env.APP_URL)
   }
```
you could take that even a little further and also abstract the boolean expression in the if statement to something like `function isS3Configured`